### ISSUE TITLE
Gran Turismo 4: Resolution upscale

### ISSUE THREAD

**Issue by lightningterror** [PCSX2/pcsx2#1796]
***PCSX2 version:***
v1.5.0-dev-1856-g72d54d3

***PCSX2 options:***
default options , mtvu on , Clamping (EE Full) (VU Extra) tho they don't seem to matter

***Plugins used:***
Default plugins

CRC Full , D3D / OpenGL Hardware 

***Description of the issue:***
Resolution is upscaled incorrectly. It's like it's upscaled and  downscaled at the same time , like in the picture below you can see it looks really pixelated , increasing the upscale to 4x and above bypasses but doesn't fix the issue ..that is because I think the resolution is downscaled only by 3x or 3.5x 
example 5x upscale -3x downscale you still get some better results
It the most visible on 2x resolution or custom resolution above native but below 2x.
I suspect for 3x and above you need a bigger monitor to see the differences

Changing the resolution to native or switching to software mode is an actual fix 

***How to reproduce the issue:***
Multiplayer split screen , choose cars for both racers and you will see the bug appear at the Begin Race menu.
Select any upscaled resolution or custom resolution in Hardware mode
All Hardware modes are affected

I think I've seen the bug appear on other places as well but I'm not sure now. This is the only location I got atm.

![gsdx_20170128012527](https://cloud.githubusercontent.com/assets/18107717/22392561/fc639024-e4fa-11e6-8233-9577feed762b.png)

gs dump
https://drive.google.com/open?id=0B8cQAq-Ko9-uZFNJNVg4OVZvb2c

***Last known version to work:***
None

***PC specifications:***
I3 4160 , 16gb ram , amd R7 360 , Windows 7 64 bit

**Comment by MrCK1** [comment]
GT4 uses alot of sprites. Texture filtering can help with pixelated sprites, but otherwise it's running as intended.

**Comment by lightningterror** [comment]
Hmm it really doesn't look that way , check the differences below , the tables and buttons seem positioned differently as well

Native
![pcsx2 2017-01-28 02-27-31-16](https://cloud.githubusercontent.com/assets/18107717/22393099/cdf4ed58-e501-11e6-9e6c-fe105d63b89d.jpg)

2x Upscale
![pcsx2 2017-01-28 02-25-47-72](https://cloud.githubusercontent.com/assets/18107717/22393102/d768bed2-e501-11e6-90c1-610227c3f2dc.jpg)

**Comment by MrCK1** [comment]
I remember seeing this a long time ago, but I've finally narrowed down the issue. The menus are comprised of many sprites that form the text you see. The blurring is caused by texture filtering.

Think of the menu as a Photoshop project composed of layers with a foreground/background. The top and bottom of the frame are in the foreground while the center is the background layer. When the picture is upscaled, the 3D model is pixelated because there is an invisible texture on top that you can't see.

TL;DR: You can fix this issue by using either:
- Bilinear (Forced) 
- Trilinear (Forced Bilinear) 
- Trilinear (Ultra/Slow)

Normal screen (Bilinear PS2):
![gsdx_20170129165241](https://cloud.githubusercontent.com/assets/16314399/22408213/533a53da-e643-11e6-9d7f-efc807a6336c.png)

Result with Bilinear (Forced) Trilinear (Forced Bilinear) or Trilinear (Ultra/Slow)
![gsdx_20170129164824](https://cloud.githubusercontent.com/assets/16314399/22408202/21b631c6-e643-11e6-8f31-40b61c5c79de.png)

**Comment by FlatOutPS2** [comment]
You're talking about a different issue. ;)

**Comment by MrCK1** [comment]
Really? Lol I thought they were the same thing :P

**Comment by FlatOutPS2** [comment]
No, but they do look similar. Your issue can also be resolved(with better results) by using MSAA. :)

**Comment by lightningterror** [comment]
Setting it to forced makes the picture blurry instead of pixelated. The issue is roughly the same because it is an upscale issue.

![pcsx2 2017-01-30 01-15-54-48](https://cloud.githubusercontent.com/assets/18107717/22409483/9ff7dc64-e68a-11e6-822d-16f18ef1f2cf.jpg)

**Comment by FlatOutPS2** [comment]
This is the issue at 2x native:
![argh](https://cloud.githubusercontent.com/assets/15942346/22419192/a7df7a52-e6dc-11e6-8f07-801872ff5521.png)

This is what it's supposed to look like:
![normal](https://cloud.githubusercontent.com/assets/15942346/22419196/aa73c2dc-e6dc-11e6-9e75-b9f5b80468ed.png)

**Comment by lightningterror** [comment]
Hmmm , the pixelating is fixed but it's still not quite right. The button on the left is off place and the car info table as well.

**Comment by FlatOutPS2** [comment]
In the snapshot none as you can tell. But I believe 500 Y and 100 X looked about right when I tried it earlier.

**Comment by RedPanda4552** [comment]
I think I've got the same issue when I run GT4 as well. Running 2x native with no offset:
![image](https://cloud.githubusercontent.com/assets/6377490/22695628/5d2372aa-ed19-11e6-8c9b-95355e3dcf85.png)
And again 2x but with 100 offset on x and 500 on y (If you look closely, yes I changed rendering engines. Had no effect on quality):
![image](https://cloud.githubusercontent.com/assets/6377490/22695649/708aef9e-ed19-11e6-9610-cfb0acfa25dc.png)
Turning on texture filtering, either Bilinear forced or Trilinear (Bilinear forced), regardless of offset settings, produces something like this:
![image](https://cloud.githubusercontent.com/assets/6377490/22695793/fa46a2fa-ed19-11e6-95be-63e99f76f442.png)

**Comment by lightningterror** [comment]
Update: This also appears in B spec mode viewing the cars from a distance , sometimes they appear very pixelated.

**Comment by lindquest** [comment]
While trying to reproduce the issue, I found that it doesn't happen if you do a clean boot of the game and don't change any of the video settings (just opening the plugin screen without changing any settings and pressing "OK", or changing to software rendering and then back with F9 counts as a change).

However, I also found that the weirdest thing happens when loading a saved state (with or without MTVU):

- If this issue occurs, any state you load after it will have the issue too (during the same run of the game). 

- If you do a clean boot of GT4 after the problem, it will happen if you load a state before the "Sony Computer Entertainment America Presents" screen appears (I have the NTSC version). 

- If you load the state right away after a clean boot, the issue won't occur if the earlier run didn't have that.

- However, it WILL happen if you load right away when it is the first run of the PCSX2 program (!).

"Clean" run (3x resolution):
!["Clean" run](https://user-images.githubusercontent.com/9299989/29246631-3a63d402-7fd7-11e7-9917-f6d41a750bdc.png)

"Dirty" run (3x):
!["Dirty" run](https://user-images.githubusercontent.com/9299989/29246632-42eeb272-7fd7-11e7-9ccf-fbc11a36eaa0.png)

Same settings as @lightningterror's, but running v1.5.0-dev-2149-g66a87ce4d on Windows 10 x64 (i7 4720HQ, GTX 970M 3 GB, 16 GB DDR3).

**Comment by lightningterror** [comment]
I'll close this since it's actually caused by upscaling.

**Comment by MrCK1** [comment]
Who hold on. Test this with the new PR, at least see if the scaling has improved.

**Comment by lightningterror** [comment]
It's still there. Thing is on the menu it's fine for a second and then you can see it switch/ the menus will move to a different spot.

I can leave it open if that's the case :smile:

**Comment by MrCK1** [comment]
You didn't need to reopen it, it was OK(leave it now) Take a look the VRAM Spike and AMD stuff ;)

**Comment by FlatOutPS2** [comment]
That is no real workaround. The issue is related to the framebuffer size(and 8bits texture unscaling I believe). An early logo just so happens to briefly use a large framebuffer size which can avoid the issue. The framebuffer size never gets reduced as it only updates when the current value is higher, but it will be reset when you change GSdx settings/press F9.

**Comment by d4d385** [comment]
Hello everybody,
when I am playing GT4, the view of the car changes randomly on its own, I should control the camera view with the select of my PS1 pad, but the system changes it for me every minutes. Does that happens to you guys as well?
Thank you
